src/node.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include<iostream>
+
+template<class ValueType>
+class Node
+{
+private:
+    ValueType value;
+
+public:
+    Node* parent;
+    Node* left;
+    Node* right;
+
+    Node(ValueType v);
+    Node(ValueType v, Node<ValueType>* p);
+    ~Node() = default;
+
+    ValueType getValue();
+    void setValue(ValueType v);
+
+    friend std::ostream& operator<<(std::ostream& str, Node<ValueType>& n) {
```

Comment:
Pytanie ogólne, czy robimy using namespace std czy taki zapis jak tu? 
Jak taki zapis, to trzeb go przyjąć dla całej aplikacji.

Reply:
W c++ w plikach nagłówkowych nie należy ustalać namespace, bo mogą być dołączone do różnych projektów i mogłoby to zakryć używaną przestrzeń nazw. Ale w głównym pliku używa się wybranej - u nas std.